src/utils.cpp

```diff
@@ -109,3 +130,163 @@ KeyRingReceive ZypperUtils::keyReport()
     return s_keyReceiveReport;
 }
 
+/************************************* ZYPPER INSTALL ************************************/
+void ZypperUtils::markPackagesForInstallation( const string& repoUrl, const string& packageName )
+{
+    Pathname sysRoot( "/" );
+    // acquire initial zypp lock
+    s_zypp = getZYpp(); 
+    
+    makeNecessaryChangesToRunAsRoot( sysRoot );
+    initTarget( sysRoot );
+    initRepoManager( repoUrl, packageName );
+    
+    //TRUST_AND_IMPORT_KEY - we are adding this to the system repos
+    KeyRing::setDefaultAccept( KeyRing::TRUST_AND_IMPORT_KEY );
+    cout << "TRUST_AND_IMPORT_KEY accepted" << endl;
+    refreshRepoManager();
```

Comment:
`TRUST_AND_IMPORT_KEY` worries me. Imported keys are not _per repo_, so the key will unlock all packages in all repos which are signed by it. If you silently accept a key here, it should IMO be asserted that the user is notified and aware of it and/or has the choice to add the key just temporarily in case it is not yet known to the system. 

Independently from that, ` KeyRing::setDefaultAccept` sets a global default __for all unknown keys__ occurring in any context. This is probably not what you want (or should do). If you want to (silently) accept the key introduced by a specific one-click repo, you should consider creating a small RAII class to provide a scoped and exception safe save-set-and-restore of the DefaultAccept value. Use it only in the scope that actually refreshes your  specific one-click repo, as this is (most probably) the only action where silently accepting a key is desired.